src/platform_impl/windows/mod.rs

```diff
@@ -235,17 +235,26 @@ impl Menu {
             let child_ = child.borrow();
             let item_type = child_.item_type();
 
+            // Set icons for both regular menu items and submenus
             if matches!(item_type, MenuItemType::Icon | MenuItemType::Submenu) {
                 let hbitmap = child_
                     .icon
                     .as_ref()
                     .map(|i| unsafe { i.inner.to_hbitmap() })
                     .unwrap_or(std::ptr::null_mut());
+
                 let info = create_icon_item_info(hbitmap);
 
                 unsafe {
-                    SetMenuItemInfoW(self.hmenu, child_.internal_id, FALSE, &info);
-                    SetMenuItemInfoW(self.hpopupmenu, child_.internal_id, FALSE, &info);
+                    // For submenus, use hmenu as ID, for regular items use internal_id
+                    let id = if matches!(item_type, MenuItemType::Submenu) {
+                        child_.hmenu as _
+                    } else {
+                        child_.internal_id()
+                    };
+
+                    SetMenuItemInfoW(self.hmenu, id, FALSE, &info);
+                    SetMenuItemInfoW(self.hpopupmenu, id, FALSE, &info);
```

Comment:
We have a method to handle this `.internal_id()`

```suggestion
                    SetMenuItemInfoW(self.hmenu, child_.internal_id(), FALSE, &info);
                    SetMenuItemInfoW(self.hpopupmenu, child_.internal_id(), FALSE, &info);
```

The naming of `.internal_id()` so if you want, feel free to change it to anything else.